Fix internal spec links (#85)
awslabs/aws-encryption-sdk-specification: client-apis/encrypt.md

### Encryption Context

See [encryption context](../framework/structures.md#encryption-context).

> The encryption context MUST NOT contain an entry with a [reserved key value](data-structutes.md#encryption-context).

Reviewer: ```suggestion
The encryption context MUST NOT contain an entry with a [reserved key value](../framework/structures.md#encryption-context).
```

This behavior MAY output the following:

- [keyring trace](../framework/structures.md#keyring-trace)

To construct the outputs, some fields MUST be constructed using information obtained
from a set of valid [encryption materials](../framework/structures.md#encryption-materials).
This behavior MUST obtain this set of [encryption materials](../framework/structures.md#encryption-materials)
by calling [Get Encryption Materials](../framework/cmm-interface.md#get-encryption-materials) on a [CMM](../framework/cmm-interface.md).
The CMM used MUST be the input CMM, if supplied.
If instead the user supplied a [keyring](../framework/keyring-interface.md), this behavior MUST use a [default CMM](../framework/default-cmm.md),
constructed using the user supplied keyring as input.
The call to [Get Encryption Materials](../framework/cmm-interface.md#get-encryption-materials) MUST include the
input [encryption context](#encryption-context), if supplied.
If the length is known on the input [plaintext](#plaintext), this call MUST also include that value

The [algorithm suite](../framework/algorithm-suites.md) used in all aspects of this behavior MUST be the algorithm suite in the

> [encryption materials](../framework/structures#encryption-materials.md) returned from the [Get Encryption Materials](../framework/cmm-interface.md#get-encryption-materials) call.

Reviewer: ```suggestion
[encryption materials](../framework/structures.md#encryption-materials) returned from the [Get Encryption Materials](../framework/cmm-interface.md#get-encryption-materials) call.
```

then the derived data key is the same as the plaintext data key.

### Message

The output message MUST be bytes, as specified by the [message format](../data-format/message.md).

The [message header](../data-format/message-header.md) is serialized with the following specifics:

- [Version](../data-format/message-header.md#version): MUST have a value corresponding to [1.0](../data-format/message-header.md#supported-versions)
- [Type](../data-format/message-header.md#type): MUST have a value corresponding to [Customer Authenticated Encrypted Data](../data-format/message.md#supported-types)
- [Algorithm Suite ID](../data-format/message-header.md#algorithm-suite-id): MUST be the [algorithm suite](../framework/algorithm-suites.md)
  used in this behavior
- [AAD](../data-format/message-header.md#aad): MUST be the serialization of the [encryption context](../framework/structures.md#encryption-context)
  in the [encryption materials](../framework/structures.md#encryption-materials)
- [Encrypted Data Keys](../data-format/message-header.md#encrypted-data-key): MUST be the serialization of the

>   [encrypted data keys](../framework/structures.md#encrypted-data-keys) in the [encryption materials](../framework/structures.md#encryption-materials)

Reviewer: ```suggestion
- [Encrypted Data Keys](../data-format/message-header.md#encrypted-data-key-entries): MUST be the serialization of the
  [encrypted data keys](../framework/structures.md#encrypted-data-keys) in the [encryption materials](../framework/structures.md#encryption-materials)
```

> - [Content Type](../data-format/message-header.md#content-type): MUST be `Framed`

Reviewer: For consistency:

```suggestion
- [Content Type](../data-format/message-header.md#content-type): MUST be [01](../data-format/message-header.md#supported-content-types)
```

> - [Content Type](../data-format/message-header.md#content-type): MUST be [01](../data-format/message-header.md#supported-content-types)

Reviewer: Sorry, that was my fault: misremembered values.
```suggestion
- [Content Type](../data-format/message-header.md#content-type): MUST be [02](../data-format/message-header.md#supported-content-types)
```

  - The AAD is the serialized [message body AAD](../data-format/message-body-aad.md)
  - The IV is the [IV](../data-format/message-body.md#iv) specified for this frame above.
  - The cipherkey is the derived data key
  - The plaintext contains part of the input [plaintext](#plaintext) this frame is encrypting.
- [Authentication Tag](../data-format/message-body.md#authentication-tag): MUST be the authentication tag outputted by the above encryption.

If the [algorithm suite](../framework/algorithm-suites.md) contains a [signature algorithm](../framework/algorithm-suites.md#signature-algorithm),
the output [message](../data-format/message.md) MUST contain a [message footer](../data-format/message-footer.md).
The footer is serialized with the following specifics:

- [Signature](../data-format/message-footer.md#signature): MUST be the output of the [signature algorithm](../framework/algorithm-suites.md#signature-algorithm)
  specified by the [algorithm suite](../framework/algorithm-suites.md), with the following input:
  - the signature key is the [signing key](../framework/structures.md#signing-key) in the [encryption materials](../framework/structures.md#encryption-materials)

>   - the input to sign is the concatenation of the serialization of the [message header](#message-header) and [message body](#message-body)

Reviewer: ```suggestion
  - the input to sign is the concatenation of the serialization of the [message header](../data-format/message-header.md) and [message body](../data-format/message-body.md)
```

Any data that is not specified within the [message format](../data-format/message.md) MUST NOT be added to the output message.

### Keyring Trace

> The [keyring trace](../framework/structures.md#keyring-trace) obtained from the [encryption materials](data-structure.md#encryption-materials).

Reviewer: ```suggestion
The [keyring trace](../framework/structures.md#keyring-trace) obtained from the [encryption materials](../framework/structures.md#encryption-materials).
```

> Implementations of the AWS Encryption SDK MUST not encrypt using the Non-Framed content type.

Reviewer: ```suggestion
Implementations of the AWS Encryption SDK MUST NOT encrypt using the Non-Framed content type.
```

However, this behavior was supported in the past.

If a message has the [non-framed](../data-format/message-body.md#non-framed-data) content type,
the [message body](../data-format/message-body.md) was serialized with the following specifics:

- [IV](../data-format/message-body.md#iv):  MUST be the [sequence number](../data-format/message-body-aad.md#sequence-number)
  used in the [message body AAD](../data-format/message-body-aad.md).
- [Encrypted Content](../data-format/message-body.md#encrypted-content): MUST be the output of the [authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
  specified by the [algorithm suite](../framework/algorithm-suites.md), with the following inputs:
  - The AAD is the serialized [message body AAD](../data-format/message-body-aad.md)
  - The IV is the [IV](../data-format/message-body.md#iv) specified above.
  - The cipherkey is the derived data key

>   - The plaintext is the input [plaintext message](#plaintext-message)

Reviewer: ```suggestion
  - The plaintext is the input [plaintext](#plaintext)
```
